examples/sync/src/bin/server.rs

```diff
@@ -0,0 +1,438 @@
+                .default_value("100"),
+        )
+        .arg(
+            Arg::new("storage-dir")
+                .short('d')
+                .long("storage-dir")
+                .value_name("PATH")
+                .help("Storage directory for database")
+                .default_value("/tmp/adb_sync_server"),
+        )
+        .get_matches();
+
+    let config = ServerConfig {
+        port: matches
+            .get_one::<String>("port")
+            .unwrap()
+            .parse()
+            .unwrap_or_else(|e| {
+                eprintln!("❌ Invalid port: {e}");
+                std::process::exit(1);
+            }),
+        initial_ops: matches
+            .get_one::<String>("initial-ops")
+            .unwrap()
+            .parse()
+            .unwrap_or_else(|e| {
+                eprintln!("❌ Invalid initial operations count: {e}");
+                std::process::exit(1);
+            }),
+        storage_dir: matches
+            .get_one::<String>("storage-dir")
+            .unwrap()
+            .to_string(),
+    };
+
+    info!("ADB Sync Server starting");
+    info!(
+        port = config.port,
+        initial_ops = config.initial_ops,
+        storage_dir = %config.storage_dir,
+        "Configuration"
+    );
+
+    let executor = tokio_runtime::Runner::default();
+    executor.start(|context| async move {
+        // Create and initialize database
+        let db_id = generate_db_id(&context);
+        let db_config = create_adb_config(&db_id);
+
+        info!(db_id = %db_id, "Initializing database");
+
+        let mut database = match Database::init(context.clone(), db_config).await {
+            Ok(db) => db,
+            Err(e) => {
+                error!(error = %e, "❌ Failed to initialize database");
+                return;
+            }
+        };
+
+        // Create and add initial operations
+        let initial_ops = create_test_operations(config.initial_ops, 12345);
+        info!(operations_len = initial_ops.len(), "Creating initial operations");
+
+        if let Err(e) = add_operations(&mut database, initial_ops).await {
+            error!(error = %e, "❌ Failed to add initial operations");
+            return;
+        }
+
+        // Commit the database to ensure operations are persisted
+        if let Err(e) = database.commit().await {
+            error!(error = %e, "❌ Failed to commit database");
+            return;
+        }
+
+        // Display database state
+        let mut hasher = Standard::new();
+        let root_hash = database.root(&mut hasher);
+        let root_hash_hex = root_hash
+            .as_ref()
+            .iter()
+            .map(|b| format!("{b:02x}"))
+            .collect::<String>();
+
+        info!(
+            op_count = database.op_count(),
+            root_hash = %root_hash_hex,
+            "Database ready"
+        );
+
+        // Create listener to accept connections
+        let addr = SocketAddr::from(([127, 0, 0, 1], config.port));
```

Comment:
I'd typically recommend this format: https://github.com/commonwarexyz/alto/blob/3bcb093342b32fd1c19733516419d905fc5e67eb/chain/src/bin/validator.rs#L89-L90

Reply:
Personally, I think `SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), config.port)` is a bit clunkier.

I'll change this to `Ipv4Addr::LOCALHOST` though.